source/NVDAObjects/UIA/winConsoleUIA.py

```diff
@@ -64,36 +61,50 @@ def move(self, unit, direction, endPoint=None):
 			if res != 0:
 				return direction
 			else:
-				if self.move(textInfos.UNIT_CHARACTER, -1):
+				if self.move(textInfos.UNIT_CHARACTER, -1): # Reset word boundaries to move to the previous word
 					return self.move(unit, direction, endPoint=endPoint)
 				else:
 					return res
-		return super(consoleUIATextInfo, self).move(unit, direction, endPoint)
+		res = super(consoleUIATextInfo, self).move(unit, direction, endPoint)
+		if oldRange and (
+			self._rangeObj.CompareEndPoints(
+				UIAHandler.TextPatternRangeEndpoint_Start,
+				firstVisiRange,
+				UIAHandler.TextPatternRangeEndpoint_Start
+			) < 0
+			or self._rangeObj.CompareEndPoints(
+				UIAHandler.TextPatternRangeEndpoint_Start,
+				lastVisiRange,
+				UIAHandler.TextPatternRangeEndpoint_End
+			) >= 0
+		):
+			self._rangeObj = oldRange
+			return 0
+		return res
 
 	def expand(self, unit):
 		if unit == textInfos.UNIT_WORD:
-			self._expandedToWord = True
+			# UIA doesn't implement word movement, so we need to do it manually.
+			offset = self._getCurrentOffsetInThisLine()
+			start, end = self._getWordOffsetsInThisLine(offset)
+			wordEndPoints = (
+				(offset - start) * -1,
+				end - offset - 1
+			)
+			self._rangeObj.MoveEndpointByUnit(
+				UIAHandler.TextPatternRangeEndpoint_Start,
+				UIAHandler.NVDAUnitsToUIAUnits[textInfos.UNIT_CHARACTER],
+				wordEndPoints[0]
```

Comment:
What happens if wordEndPoints[0] or wordEndPoints[1] is 0? I.e. you don't ned to move? perhaps these moveByUnit calls should be within an if conndition checking that wordEndPoints[x] is not 0.